### ISSUE TITLE
Queue rearrange API

### ISSUE THREAD

**Issue by space-nuko** [comfyanonymous/ComfyUI#684]
It would be cool to have an API so that you could drag around items on the queue and the order would be updated on the backend, in addition to the deletion API already available

**Comment by comfyanonymous** [comment]
You can insert items in any position in the queue using the optional 'number' property that you can put in the json you send in the /queue post request.

The queue in the backend is a priority queue sorted by that number.

**Comment by space-nuko** [comment]
Could you decide to change the priority after the prompt is set though

**Comment by comfyanonymous** [comment]
Yeah I could add an endpoint where you post the prompt id with a new number.

**Comment by space-nuko** [comment]
Preferably in such a way you could insert an existing prompt between two other prompts like number 6 and 7, so that there's no overlap for the numbers, thanks

**Comment by comfyanonymous** [comment]
the numbers are treated as floats so to insert a prompt between 6 and 7 you just have to use 6.5 or whatever number in between.

**Comment by n0valis** [comment]
> You can insert items in any position in the queue using the optional 'number' property that you can put in the json you send in the /queue post request.
> 
> The queue in the backend is a priority queue sorted by that number.

Can you show me the documentation for that please. I woluld like do that, but from your statement that is impossible.
Thanks.
